app/src/main/java/com/gdsctsec/smartt/database/Database.kt

```diff
@@ -0,0 +1,35 @@
+package com.gdsctsec.smartt.database
+
+import android.content.Context
+import androidx.room.Database
+import androidx.room.Room
+import androidx.room.RoomDatabase
+
+@Database(entities = [TimeTable::class], version = 1, exportSchema = false)
+abstract class LectureDatabase : RoomDatabase(){
+
+    abstract fun lectureDao(): LectureDao
+
+    companion object{
+        @Volatile
+        private var INSTANCE: LectureDatabase? = null
+
+        fun getInstance(context: Context): LectureDatabase {
```

Comment:
from where will this `getInstance()` be called?

Reply:
I think it will be in VM

Reply:
if you want the database to be created as soon as the app is installed, then here is the code:
`companion object {
        @Volatile
        private var instance: AppDatabase? = null
        private val LOCK = Any()

        operator fun invoke(context: Context) = instance ?: synchronized(LOCK) {
            instance ?: buildDatabase(context).also {
                instance = it
            }
        }

        private fun buildDatabase(context: Context) =
            Room.databaseBuilder(
                context.applicationContext,
                AppDatabase::class.java,
                "database.db"
            ).build()
    }`

Reply:
The current code does the same, just syntax is different